app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -49,18 +50,25 @@ export default function*() {
   ];
 }
 
-function* createPageSession({ id, location, options }) {
+function* createPageSession({ id, location, options, withUI = true }) {
```

Comment:
Indeed, a bulk action might be a better idea in this particular case. For `close all tabs` and `close other tabs`, composing multiple `destroyPageSession` actions together is straightforward, no need for any custom shenanigans.